core/blockchain.go

```diff
@@ -1168,6 +1170,9 @@ func (bc *BlockChain) WriteBlockWithState(
 						bc.triegc.Push(root, number)
 						break
 					}
+					if -number > bc.maxGarbCollectedBlkNum {
```

Comment:
why is number negative?

Reply:
b/c its a prio queue, key by negative block nums. Notice the push here: 

https://github.com/Daniel-VDM/harmony/blob/834e09bf03a3ce0cda83866ee50755d5a4118b28/core/blockchain.go#L1134